### ISSUE TITLE
Null Pointer exception while trying to close connection

### ISSUE THREAD

**Issue by Nbhaz** [brettwooldridge/HikariCP#930]
#### Environment
```
HikariCP version: 2.6.2
JDK version     : 1.8.0_111
Database        : MySQL
Driver version  : 5.1.38
```
-----------------------------------------------------------------------------------------
Getting following exceptions: 

2017-07-07 09:45:58.040 IST DEBUG rid=NA cid=NA 2717 --- [nnection closer] com.zaxxer.hikari.pool.PoolBase          : mdsDataSourcePool - Closing connection com.mysql.jdbc.JDBC4Connection@610dffca: (connection has passed maxLifetime)
Exception in thread "mdsDataSourcePool network timeout executor" java.lang.NullPointerException
	at com.mysql.jdbc.MysqlIO.setSocketTimeout(MysqlIO.java:4980)
	at com.mysql.jdbc.ConnectionImpl$12.run(ConnectionImpl.java:5529)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by brettwooldridge** [comment]
Are you sure your driver version is correct?  [The lines numbers don't match.](https://github.com/mysql/mysql-connector-j/blob/5.1.38/src/com/mysql/jdbc/MysqlIO.java)

**Comment by brettwooldridge** [comment]
Looks like you might be running [v5.1.40](https://github.com/mysql/mysql-connector-j/blob/5.1.40/src/com/mysql/jdbc/MysqlIO.java#L4980).  We reported this issue to MySQL two or three years ago -- it is fixed in [v5.1.41](https://github.com/mysql/mysql-connector-j/blob/5.1.41/src/com/mysql/jdbc/MysqlIO.java#L4983), but I recommend using v5.1.42 or newer.
